library/nbde_client_clevis.py

```diff
@@ -1419,21 +1423,26 @@ def bindings_sanity_check(bindings, data_dir, check_mode):
             errmsg = "Each binding must have a device set"
             return None, {"msg": errmsg}
 
-        # When running under check mode, key_file is not used, which means we
+        # When running under check mode, encryption_key is not used, which means we
         # also do not need to have data_dir defined.
-        if "key_file" in binding and not check_mode:
+        if (
+            "encryption_key" in binding or "encryption_key_src" in binding
```

Comment:
We are sanitizing the input we got from the role here. Until now, we were only expecting `encryption_key_src`, which is a possible dictionary key in `nbde_client_bindings`, that we get from the role. Based on it, we create `encryption_key` accordingly, using the key file which should be in`data_dir`, to be used by the module. If we want to be able to receive `encryption_key` from the role, we should also document it in the `README`, no?

By the way, I noticed that in our module documentation, above, we are naming the option `nbde_client_bindings`, but it should be `bindings`. The module receives two parameters, `bindings` and `data_dir`. `nbde_client_bindings` is how the role receives the data.

Reply:
The problem here with `encryption_key` and `encryption_key_src` is that, at the role level, `encryption_key_src` is correct - it is the file on the control node that the role will copy to the managed nodes.
```yaml
nbde_client_bindings:
  - encryption_key_src: /path/on/control/node/file.key
```
However, the module expects a parameter which is the name of the key file on the managed node.  And `nbde_client_bindings` gets passed as-is to the module:
```
    - name: Perform clevis operations
      nbde_client_clevis:
        bindings: "{{ nbde_client_bindings | default([]) }}"
        data_dir: "{{ nbde_client_tempdir.path }}"
      ...
```
that is, it looks like this when expanded:
```
    - name: Perform clevis operations
      nbde_client_clevis:
        bindings:
          - encryption_key_src: /path/on/control/node/file.key
        data_dir: "{{ nbde_client_tempdir.path }}"
      ...
```
The other wrinkle is that the module ignores the directory part of `encryption_key_src`.  It looks for the file at `{{ data_dir }}/{{ basename encryption_key_src }}`

If we really wanted to implement the strict tls/crypto parameter naming semantic at the module level, the role would have to convert the argument by creating a new `encryption_key` key from the `:
```
for each binding in bindings:
   binding.encryption_key = {{ data_dir }}/{{ basename binding.encryption_key_src }}
   delete binding.encryption_key_src
```
And then `binding.encryption_key` would be the full path and file name of the key file on the managed node.  Alternately
```
for each binding in bindings:
   binding.encryption_key = {{ basename binding.encryption_key_src }}
   delete binding.encryption_key_src
```
And then `binding.encryption_key` would be the file name of the key file on the managed node relative to `data_dir`.

That's pseudo code - would have to figure out how to implement that in jinja2.

But - IMO - since the module is not part of the public interface, the module can interpret the `binding.encryption_key_src` as it wants, and can interpret it as "This argument is the name of the file on the control node, but just use the base name of this file relative to `data_dir`".  Then we can document this in the module, and I can remove references to `encryption_key` in the module docs.

@pcahyna @sergio-correia ok?

Reply:
Yes, while it is not an especially nice interface (the one of the module), it is not public, so we can keep it. I would not bother refactoring this now.